transactions/src/test/java/org/jboss/hal/testsuite/test/transactions/TransactionCommitAndRollbackTestCase.java

```diff
@@ -0,0 +1,248 @@
+            "AAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASbwAAAAEAAAAAAAAAAAAAAAAAAP__CigFZuOgfONXxVvAAAASbjEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFrgAAAAEAAAAAAAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFrTEHAgIA",
+            "AAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFsgAAAAEAAAAAAAAAAAAAAAAAAP__CigFZtXWO0pXxVwMAAAFrTEHAgIA"
+    };
+
+    private static final Vector<PreparedTransactionTestCell> TRANSACTIONS_TEST_DATA = new Vector<PreparedTransactionTestCell>() {
+        {
+        add(new PreparedTransactionTestCell(TRANSATIONS_XID[0], PerformedAction.ROLLBACK));
+        add(new PreparedTransactionTestCell(TRANSATIONS_XID[1], PerformedAction.COMMIT));
+        add(new PreparedTransactionTestCell(TRANSATIONS_XID[2], PerformedAction.COMMIT));
+        add(new PreparedTransactionTestCell(TRANSATIONS_XID[3], PerformedAction.ROLLBACK));
+        add(new PreparedTransactionTestCell(TRANSATIONS_XID[4], PerformedAction.COMMIT));
+        add(new PreparedTransactionTestCell(TRANSATIONS_XID[5], PerformedAction.ROLLBACK));
+        add(new PreparedTransactionTestCell(TRANSATIONS_XID[6], PerformedAction.COMMIT));
+        }
+    };
+
+    @Test
+    public void testTransactionsRollbackAndCommit() throws InterruptedException, ExecutionException, TimeoutException {
+        final ExecutorService executors = Executors.newSingleThreadExecutor();
+        final LogListenerForRollbackAndCommit listener = new LogListenerForRollbackAndCommit();
+        final Tailer tailer = new Tailer(serverLog.toFile(), listener, 1);
+        Future tailerFuture = null;
+        try {
+            logger.debug("Starting tailer to monitor logfile!");
+            tailerFuture = executors.submit(tailer);
+            for (PreparedTransactionTestCell transactionTestCell : TRANSACTIONS_TEST_DATA) {
+                page.selectPreparedTransaction(transactionTestCell.getXId());
+                page.clickButton(transactionTestCell.getPerformedAction().getButtonText());
+            }
+        } finally {
+            logger.debug("Stopping tailer!");
+            tailer.stop();
+            if (tailerFuture != null) {
+                tailerFuture.get(1, TimeUnit.SECONDS);
+            }
+            executors.shutdown();
+        }
+
+        //check if finished successfully
+        List<AssertionError> errors = new LinkedList<>();
+        for (PreparedTransactionTestCell transactionTestCell : TRANSACTIONS_TEST_DATA) {
+            try {
+                Assert.assertTrue(transactionTestCell.getFinishedSuccessfully());
+            } catch (AssertionError error) {
+                errors.add(error);
+            }
+        }
+        if (errors.size() > 0) {
+            throw new AssertionError(errors.stream().map(AssertionError::getMessage).collect(Collectors.joining(", ")));
+        }
+    }
+
+    private static final class LogListenerForRollbackAndCommit extends TailerListenerAdapter {
+
+        public LogListenerForRollbackAndCommit() {
+        }
```

Comment:
Empty constructor exists by default if there is no other constructor if I am not mistaken, if it is so, please remove it.